### PR TITLE
Fix for exception thrown if no delimiters were specified

### LINKED ISSUE #6624: exception when filling new group dialog wrongly
JabRef 5.1--2020-06-17--ef0a6bb
Linux 5.6.18-300.fc32.x86_64 amd64 
Java 14.0.1

- [x] **Mandatory**: I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. add subgroup
2. collect by keywords
3. generate groups by ... enter "field=specifier"

I recognise this may not be the content expected. 

Solution: An error message that tells the user what input would be ideal.

Screenshot-20200620135115-620x495

<details>
  <summary>Log File</summary>
  
  ```
java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.base/java.lang.StringLatin1.charAt(Unknown Source)
	at java.base/java.lang.String.charAt(Unknown Source)
	at org.jabref/org.jabref.gui.groups.GroupDialogViewModel.resultConverter(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.control.Dialog.setResultAndClose(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.control.DialogPane.lambda$createButton$3(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.Node.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.control.Button.fire(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.scene.control.inputmap.InputMap.handle(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.gtk.GtkApplication.enterNestedEventLoopImpl(Native Method)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.gtk.GtkApplication._enterNestedEventLoop(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.Application.enterNestedEventLoop(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.EventLoop.enter(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.stage.Stage.showAndWait(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.control.HeavyweightDialog.showAndWait(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.control.Dialog.showAndWait(Unknown Source)
	at org.jabref/org.jabref.gui.JabRefDialogService.showCustomDialogAndWait(Unknown Source)
	at org.jabref/org.jabref.gui.groups.GroupTreeViewModel.lambda$addNewSubgroup$7(Unknown Source)
	at java.base/java.util.Optional.ifPresent(Unknown Source)
	at org.jabref/org.jabref.gui.groups.GroupTreeViewModel.addNewSubgroup(Unknown Source)
	at org.jabref/org.jabref.gui.groups.GroupTreeView.lambda$createContextMenuForGroup$20(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.control.MenuItem.fire(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.scene.control.ContextMenuContent$MenuItemContainer.doSelect(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.scene.control.ContextMenuContent$MenuItemContainer.lambda$createChildren$12(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.1.628/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
  ```
</details>

### PR DESCRIPTION
**Pull request by alex-petrov-vt**
Assigned default values for delimiters to fix the exception thrown
Fixes #6624 

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by alex-petrov-vt** [comment]
#6642

File: src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java [hunk 1]
```diff
@@ -313,8 +313,11 @@ public AbstractGroup resultConverter(ButtonType button) {
                             groupName,
                             groupHierarchySelectedProperty.getValue(),
                             FieldFactory.parseField(autoGroupKeywordsFieldProperty.getValue().trim()),
-                            autoGroupKeywordsDelimiterProperty.getValue().charAt(0),
-                            autoGroupKeywordsHierarchicalDelimiterProperty.getValue().charAt(0));
+                            autoGroupKeywordsDelimiterProperty.getValue().isEmpty() ?
```
Comment (koppor) [inline src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java:316]: I find this code very hard to read. Could you extract if statements generating the value into a variable and then use the variable?

File: src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java [hunk 2]
```diff
@@ -309,12 +309,22 @@ public AbstractGroup resultConverter(ButtonType button) {
                         searchGroupRegexProperty.getValue());
             } else if (typeAutoProperty.getValue()) {
                 if (autoGroupKeywordsOptionProperty.getValue()) {
+                    // Set default value for delimiters: ',' for base and '>' for hierarchical
+                    char delimiter = ',';
+                    char hierarDelimiter = '>';
```
Comment (koppor) [inline src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java:314]:
```suggestion
                    char hierarDelimiter = Keyword.DEFAULT_HIERARCHICAL_DELIMITER;
```

(org.jabref.model.entry.Keyword)

**Comment by koppor** [comment]
I commited my update. The required "import" is not added. Maybe, someone can add it.  Otherwise, I'll try to find time on the weekend and go ahead with merging; this is some kind of hotfix and should go in early.

**Pull request merged**